core/unix/rseq_linux.c

```diff
@@ -75,9 +76,15 @@ typedef struct _rseq_region_t {
     app_pc start;
     app_pc end;
     app_pc handler;
-    /* XXX i#2350: Additional information will be added here for eliminating
-     * some existing limitations.
+    /* We need to preserve input registers for targeting "start" instead of "handler"
+     * for our 2nd invocation.  We only support GPR inputs.
```

Comment:
What about flags by the way?

Reply:
We assume they are not an input.  That seems pretty unlikely.  I will add an explicit comment.

Reply:
Ok!